src/sidebar/store/create-store.js

```diff
@@ -181,20 +187,38 @@ export function createStore(modules, initArgs = [], middleware = []) {
   return store;
 }
 
+// The properties of the `config` argument to `createStoreModule` below are
+// declared inline due to https://github.com/microsoft/TypeScript/issues/43403.
+
 /**
- * Helper to validate a store module configuration before it is passed to
- * `createStore`.
+ * Create a store module that can be passed to `createStore`.
  *
  * @template State
  * @template Actions
- * @template Selectors
+ * @template {SelectorMap<State>} Selectors
```

Comment:
This allows TS to relate the `state` argument of selectors to the `initialState` argument to the function. The upshot is:

```js
const initialState = { things: [] };
createStoreModule(initialState, {
  selectors: {
    // TS knows the type of `state` here automatically based on `initialState`.
    countThings(state) { return state.things.length; }
  }
});
```